src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  health = 100;
+
+  constructor(name) {
+    this.name = name;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    const checkPrototype = target instanceof Herbivore && !target.hidden;
+
+    target.health = checkPrototype
+      ? target.health - 50
+      : target.health;
+
+    Animal.alive = Animal.alive
+      .filter(animal => animal.health > 0);
+  }
```

Comment:
will be better to use if statement here
and change animal health and call filter method only if needed statement is true

Reply:
your advise was great! 🔥 
![image](https://user-images.githubusercontent.com/105124305/192151918-a763e37d-4c10-4a80-8d93-c9c543baaa26.png)